### PR TITLE
NOISSUE: enforce Any marshaling

### PR DESCRIPTION
**Pull request by ElSacramento**
check length before unmarshal

### CODE REVIEW

File: ledger-core/rms/registry.go [hunk 1]
```diff
@@ -120,6 +122,9 @@ func (p *TypeRegistry) GetSpecial(id uint64, special string) reflect.Type {
 }
 
 func UnmarshalType(b []byte, typeFn func(uint64) reflect.Type) (uint64, reflect.Type, error) {
+	if len(b) == 0 {
```
Comment (cyraxred) [inline ledger-core/rms/registry.go:125]: Wrong. Please don't do it

File: ledger-core/rms/any.go [hunk 2]
```diff
@@ -55,13 +55,21 @@ func (p *Any) MarshalTo(b []byte) (int, error) {
 	if p.value != nil {
 		return p.value.MarshalTo(b)
 	}
+	_, _, err := UnmarshalType(b, func(u uint64) reflect.Type { return reflect.TypeOf(struct{}{}) })
+	if err != nil {
+		return 0, err
+	}
 	return 0, nil
```
Comment (cyraxred) [inline ledger-core/rms/any.go:62]: why do you need if here? not just 
return 0, err

File: ledger-core/rms/any.go [hunk 3]
```diff
@@ -55,13 +55,21 @@ func (p *Any) MarshalTo(b []byte) (int, error) {
...
 
 func (p *Any) MarshalToSizedBuffer(b []byte) (int, error) {
 	if p.value != nil {
 		return p.value.MarshalToSizedBuffer(b)
 	}
+	_, _, err := UnmarshalType(b, func(u uint64) reflect.Type { return reflect.TypeOf(struct{}{}) })
+	if err != nil {
+		return 0, err
+	}
 	return 0, nil
```
Comment (cyraxred) [inline ledger-core/rms/any.go:73]: same as above

**Review by cyraxred — CHANGES_REQUESTED** [review]

File: ledger-core/rms/any.go [hunk 4]
```diff
@@ -55,13 +55,21 @@ func (p *Any) MarshalTo(b []byte) (int, error) {
 	if p.value != nil {
 		return p.value.MarshalTo(b)
 	}
+	_, _, err := UnmarshalType(b, func(u uint64) reflect.Type { return reflect.TypeOf(struct{}{}) })
```
Comment (cyraxred) [inline ledger-core/rms/any.go:58]: pls replace a local closure with a global func that returns a pre-defined value

File: ledger-core/rms/any.go (hunk 2 above)
Comment (ElSacramento) [reply ledger-core/rms/any.go]: fixed

**Review by AndreyBronin — APPROVED** [review]

**Pull request merged**